### ISSUE TITLE
IndicesIndexSettings Interface doesn't have mapping property

### ISSUE THREAD

**Issue by ubaranzorlu** [elastic/elasticsearch-js#1681]
## 🐛 Bug Report

**IndicesIndexSettings** Interface inside `lib > api > typesWithBodyKey.d.ts` doesn't have a mapping property so it is impossible to define mapping total fields limit on indices creation.

## To Reproduce

```ts
await client.indices.create({
    index: ELASTIC_INDEX,
    body: {
      mappings: {
        properties: {
         // Insert here more than 1000 fields
      },
      settings: {
        index: {
          mapping: {
            total_fields: {
              limit: 2000,
            },
          },
        },
      },
    },
  });
```

The code above should execute without any problems but it shows a type check problem "No overload matches this call."

## Expected behavior

If I append mapping property to the IndicesIndexSettings Interface by hand

```ts
export interface IndicesIndexSettings {
    mapping?: any, // Here I added
    index?: IndicesIndexSettings;
    mode?: string;
    routing_path?: string | string[];
```

It executes as I would expect in Dev Tools and accepts my document with more than 1000 properties.

## Your Environment

- *node version*: v16.3.0
- `@elastic/elasticsearch` *version*: 8.1.0
- *os*: Mac

**Comment by delvedor** [comment]
Hello! You are right, the `mapping` field is missing there. It has already been fixed in `v8.2`.
Unfortunately `v8.2` has a [bug](https://github.com/elastic/elasticsearch-js/issues/1686). A fix for that will be out soon.

**Comment by ubaranzorlu** [comment]
> Hello! You are right, the `mapping` field is missing there. It has already been fixed in `v8.2`.
> Unfortunately `v8.2` has a [bug](https://github.com/elastic/elasticsearch-js/issues/1686). A fix for that will be out soon.

Yes, it seems it is fixed in v8.2. I will be waiting for that patch, thanks.
